python/lsst/ts/hvac/hvac_csc.py

```diff
@@ -0,0 +1,283 @@
+# This file is part of ts_hvac.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["HvacCsc"]
+
+import asyncio
+import pathlib
+
+from lsst.ts import salobj
+from lsst.ts.hvac.mqtt_client import MqttClient
+from lsst.ts.hvac.simulator.sim_client import SimClient
+
+
+class HvacCsc(salobj.ConfigurableCsc):
+    """Commandable SAL Component for the HVAC (Heating, Ventilation and Air
+    Conditioning).
+
+    Parameters
+    ----------
+    config_dir : `string`
+        The configuration directory
+    initial_state : `salobj.State`
+        The initial state of the CSC
+    simulation_mode : `int`
+        Simulation mode (1) or not (0)
+    """
+
+    def __init__(
+        self, config_dir=None, initial_state=salobj.State.STANDBY, simulation_mode=0,
+    ):
+        schema_path = pathlib.Path(__file__).resolve().parents[4].joinpath("schema", "hvac.yaml")
+        self.config = None
+        self._config_dir = config_dir
+        super().__init__(
+            name="HVAC",
+            index=0,
+            schema_path=schema_path,
+            config_dir=config_dir,
+            initial_state=initial_state,
+            simulation_mode=simulation_mode,
+        )
+        self.mqtt_client = None
+        self.telemetry_task = None
```

Comment:
The task gets initialized in the connect function and then will wait for telemetry to arrive so it can be published. Is this possible with a done asyncio.Future?

Reply:
We talked about this in person, but for the public record: my suggestion is to initialize self.telemetry_task to salobj.make_done_future() rather than None. You should still set it to asyncio.create_task(self.telemetry_loop()) later, when you start the telemetry loop.

The purpose of this change is to make it so that `self.telemetry_task` is always a `Task` (or `Future`, which is essentially the same thing), never `None`. Thus you can always safely cancel it, safely ask if it's done, etc.